### PR TITLE
Add invert button to image toolbar

### LINKED ISSUE #1716: Subject image invert button
**Package**  lib-classifier

**Describe the solution you'd like**

If enabled for a workflow, this displays a toggle to CSS color invert the subject image. Currently on PFE, it's a toggle that appears below the subject image (highlighted by me with the red square):

Screen Shot 2020-07-20 at 11 17 22 AM

I didn't find an existing design for this. @beckyrother do you want this toggle to take a similar placement and design as the below image viewer options or to be a button in the image toolbar with the pan/zoom functions?

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
- closes #1716 

## Describe your changes
- adds image invert button to image toolbar
- adds `invert_subject` to workflow configuration model
- conditionally show invert button per workflow configuration
- add invert property to subject viewer model
- add invert styling to image viewer per subject viewer invert property
- refactor with react-testing-library

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=1952&workflow=3594&demo=true 
- What user actions should my reviewer step through to review this PR?
  - click Invert button, see image colors inverted
  - click different toolbar button, invert button remains actively styled (like annotation button)
  - click Invert button again, see image colors revert to original
- Which storybook stories should be reviewed?
  - Invert button - http://localhost:6006/?path=/story/image-toolbar-invertbutton--default&globals=locale:en;locales.en:English;locales.test:Test+Language
  - Image toolbar with Invert button - http://localhost:6006/?path=/story/image-toolbar-image-toolbar--default&globals=locale:en;locales.en:English;locales.test:Test+Language
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes - update the PFE FEM Lab to enable the related workflow setting checkbox

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js [hunk 1]
```diff
@@ -82,6 +83,11 @@ const SubjectViewer = types
         self.fullscreen = false
       },
 
+      invertView () {
+        console.log('invert view')
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:87]:
```suggestion
```

Just checking if this log is intentionally here - I don't think it's necessary for production.

**Review by goplayoutside3 — COMMENTED** [review]
Looks good! I'm able to invert the subject while running locally: https://local.zooniverse.org:3000/projects/markb-panoptes/digileap-testing-staging/classify/workflow/3594.

The invert css styling works on Chrome, Firefox, Edge, but not Safari (mobile and desktop). It looks like Safari does apply the intended css property, but the image isn't actually inverted.

A non-blocking suggestion for storybook - ImageToolbar's story functionality works as expected, but the highlight around the active buttons is an oval shape instead of a circle. The zoom-in and zoom-out buttons seems to keep their circle shape, but the rest look off. Perhaps a follow-up PR to standardize all button styling in the toolbar?

toolbar

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
-import { Box } from 'grommet'
 import InteractionLayer from '../InteractionLayer'
 import ZoomControlButton from '../ZoomControlButton'
 
-function SingleImageViewer(props) {
+const StyledBox = styled(Box)`
+  image {
+    filter: ${({ invert }) => invert ? 'invert(100%)' : 'invert(0%)'};
+  }
+`
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js:14]:
```suggestion
```
This styling won't work in all browsers.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js [hunk 3]
```diff
@@ -27,15 +36,22 @@ function SingleImageViewer(props) {
...
+          onClick={zoomControlFn}
+          zooming={zooming}
+        />
       )}
-      <Box animation='fadeIn' overflow='hidden'>
+      <StyledBox
+        animation='fadeIn'
+        invert={invert}
+        overflow='hidden'
+      >
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js:48]:
```suggestion
      <Box
        animation='fadeIn'
        overflow='hidden'
      >
```

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js [hunk 4]
```diff
@@ -27,15 +36,22 @@ function SingleImageViewer(props) {
...
         <svg
           focusable
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js:50]:
```suggestion
        <svg
          filter={invert ? 'invert(100%)' : 'invert(0)'}
          focusable
```

Using a style attribute on the SVG element here should fix the invert button in Safari.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js [hunk 5]
```diff
@@ -46,18 +62,23 @@ function SingleImageViewer(props) {
...
+              <InteractionLayer
+                scale={scale}
+                height={height}
+                width={width}
+              />
             )}
           </g>
         </svg>
-      </Box>
+      </StyledBox>
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewer.js:73]:
```suggestion
      </Box>
```

**Review by eatyourgreens — COMMENTED** [review]
I'd suggest using a `filter` [presentation attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/Presentation) on the SVG element, rather than CSS styles. That should get this working in Safari.

**Comment by mcbouslog** [comment]
The ImageToolbar circles showing as ovals has been issue for awhile, was just discussing with Sean a few weeks ago, and I think is due to missing the grid from the default layout. I imported a few styled components, including grid styling, from DefaultLayout into the ImageToolbar story and I think it's working as expected now!

File: packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js:87]: I can remove, but included to match resetView, rotate, zoom in, and zoom out, which have similar console.logs, though I'm not entirely sure why. I can remove just invert, remove all, or keep all (as is)?

**Pull request merged**